src/Language/PureScript/Types.hs

```diff
@@ -107,24 +110,22 @@ $(A.deriveJSON A.defaultOptions ''Type)
 $(A.deriveJSON A.defaultOptions ''Constraint)
 $(A.deriveJSON A.defaultOptions ''ConstraintData)
 
--- |
--- Convert a row to a list of pairs of labels and types
---
+-- | Convert a row to a list of pairs of labels and types
 rowToList :: Type -> ([(Label, Type)], Type)
-rowToList (RCons name ty row) = let (tys, rest) = rowToList row
-                                in ((name, ty):tys, rest)
-rowToList r = ([], r)
+rowToList = go where
+  go (RCons name ty row) =
+    first ((name, ty) :) (rowToList row)
+  go r = ([], r)
 
--- |
--- Convert a list of labels and types to a row
---
+-- | Convert a row to a list of pairs of labels and types, sorted by the labels.
+rowToSortedList :: Type -> ([(Label, Type)], Type)
+rowToSortedList = first (sortBy (comparing fst)) . rowToList
```

Comment:
Is this safe if the ordering of entries with the same label matters? I guess only if sortBy is stable?